src/tmpfiles/tmpfiles.c

```diff
@@ -208,20 +216,148 @@ static const Specifier specifier_table[] = {
         {}
 };
 
+static int specifier_kernel_isol(char specifier, const void *data, const void *userdata, char **ret) {
+        if (!empty_or_root(arg_root))
+                return -ENXIO;
+
+        return specifier_kernel_release(specifier, data, userdata, ret);
+}
+
+static int specifier_arch_isol(char specifier, const void *data, const void *userdata, char **ret) {
+        if (!empty_or_root(arg_root))
+                return -ENXIO;
+
+        return specifier_architecture(specifier, data, userdata, ret);
+}
+
+static int id128_read_isol(const char *prefix, const char *p, Id128Format f, sd_id128_t *ret) {
+        _cleanup_free_ char *fullpath = NULL;
+
+        fullpath = path_join(arg_root, p);
+        if (!fullpath)
+                return -ENOMEM;
+        return id128_read(fullpath, f, ret);
+}
+
+static int specifier_machine_id_isol(
+        const char *prefix, char specifier, const void *data, const void *userdata, char **ret) {
+        int r;
+        char *n;
+        sd_id128_t machine_id = {};
+
+        r = id128_read_isol(prefix, "/etc/machine-id", ID128_PLAIN, &machine_id);
+        if (r < 0)
+                return r;
+        if (sd_id128_is_null(machine_id))
+                return -ENOMEDIUM;
+
+        n = new (char, 33);
+        if (!n)
+                return -ENOMEM;
+
+        *ret = sd_id128_to_string(machine_id, n);
+        return r;
+}
```

Comment:
hmm, so this will come up in other tools too. I think it would be better to just teach the generic specifier_machine_id() the trick instead of adding this to every tool individually. I.e. use the `data` argument to specifier_machine_id() (which we currently pass as NULL always) to pass in the root dir to operate on.

Reply:
extended a function in `id128-util.c`, and the `specifier_machine_id` function